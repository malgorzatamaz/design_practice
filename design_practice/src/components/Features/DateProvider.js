import React from "react"

const DateProvider = props => {
  const dateNow = Date.now()
  const date = {}
  date.month = dateNow.getMonth() + 1
  date.day = dateNow.getDate()
  date.year = dateNow.getFullYear()

  return <React.Fragment>{props.render(date)}</React.Fragment>
}

export default DateProvider
