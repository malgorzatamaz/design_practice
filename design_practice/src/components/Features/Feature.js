import React from "react"

import "./Feature.sass"

const Feature = ({ title, subtitle, renderIcon }) => {
  return (
    <div className="feature">
      <div className="feature__tile">{renderIcon()}</div>
      <div>
        <div className="feature__title">{title}</div>
        <div className="feature__subtitle">{subtitle}</div>
      </div>
    </div>
  )
}

export default Feature
