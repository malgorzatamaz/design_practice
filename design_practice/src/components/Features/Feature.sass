.feature
    display: flex
    flex: 0 1 50%
    padding: $m-sm

    &__icon
        border-radius: $border-radius
        height: 50px
        padding: $m-md
        width: 50px

        &.-yellow
            background-color: orange

        &.-green
            background-color: green

        &.-red
            background-color: red

        &.-blue
            background-color: blue

    &__tile
        margin: $m-sm

    &__title
        font-weight: $font-weight-bold
        padding: $m-sm 0

    &__subtitle
       font-size: $font-size-sm
