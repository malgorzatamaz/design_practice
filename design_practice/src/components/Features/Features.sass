.features
    display: flex
    flex-direction: column
    padding: $m-md 0
    max-width: 1000px

    &__link
        align-self: center
        padding: $m-lg 0

    &__content
        display: flex
        flex-wrap: wrap
        flex-direction: column

    &__title
        font-size: $font-size-xl

        b
            margin-right: $m-xs

    &__title, 
    &__subtitle
        text-align: center
        margin-bottom: $m-md


+media-breakpoint-up(xl)
    .features__content
        display: flex
        flex-wrap: wrap
        flex-direction: row
