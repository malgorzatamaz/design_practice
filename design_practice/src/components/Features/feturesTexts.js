const title1 = "Improve communication"
const title2 = "Be more productive"
const title3 = "Stay organized"
const title4 = "Collaborate effectively"

const description1 = `Pictures speak louder than words. Use that! Snap Screenshots
and mark them up. Do you need to clarify a more complex
problem? Record a Screen. Get your point across in a few clicks.`
const description2 = `Don't let the lack of information slow you down. Use Droplr
Integrations and make the flow of information seamless.
Communicate as you go and finish your projects faster`
const description3 = `Store your files in a cloud and have them always easily
accessible. Create category Boards to avoid clutter. Droplr
clean, easy to-use interface makes managing files hassle-free.`
const description4 = `Empower whole teams to collaborate. Use Droplr Dashboard
as a collaboration platform. Let others access and modify the
content of your Boards so that everyone is up-to-speed.`

export default {
  title1,
  title2,
  title3,
  title4,
  description1,
  description2,
  description3,
  description4
}
