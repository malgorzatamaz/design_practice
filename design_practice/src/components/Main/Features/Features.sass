.features
    margin: $m-md 0
    display: flex
    flex-direction: column

    a
        align-self: center
        padding: $m-lg 0

    &__content
        display: flex
        flex-wrap: wrap

    &__title
        font-size: $font-size-xl

        & .bold-text
            margin-right: $m-xs

    &__subtitle
        font-size: $font-size-base

    &__title, &__subtitle
        display: flex
        justify-content: center