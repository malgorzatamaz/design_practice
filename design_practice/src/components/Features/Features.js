import Feature from "./Feature"
import React from "react"

import { ReactComponent as CloudIcon } from "../../assets/icons/cloud.svg"
import { ReactComponent as CursorIcon } from "../../assets/icons/cursor.svg"
import { ReactComponent as RocketIcon } from "../../assets/icons/rocket.svg"
import { ReactComponent as TileIcon } from "../../assets/icons/tiles.svg"
import texts from "./feturesTexts"

import "./Features.sass"

const Features = () => {
  return (
    <div className="features">
      <div className="features__title">
        <span className="bold-text">Droplr</span> lets you
      </div>
      <div className="features__subtitle">
        Cut down on complexity and have the job done, simply
      </div>
      <div className="features__content">
        <Feature
          renderIcon={() => <CursorIcon className="feature__icon -green" />}
          title={texts.title1}
          subtitle={texts.description1}
        />
        <Feature
          renderIcon={() => <RocketIcon className="feature__icon -red" />}
          title={texts.title2}
          subtitle={texts.description2}
        />
        <Feature
          renderIcon={() => <CloudIcon className="feature__icon -blue" />}
          title={texts.title3}
          subtitle={texts.description3}
        />
        <Feature
          renderIcon={() => <TileIcon className="feature__icon -yellow" />}
          title={texts.title4}
          subtitle={texts.description4}
        />
      </div>
      <a href="#features">Browse all awesome Droplr features</a>
    </div>
  )
}

export default Features
