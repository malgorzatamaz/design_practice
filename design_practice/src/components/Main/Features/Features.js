import React from "react"
import Feature from "./Feature"

import { ReactComponent as TileIcon } from "../../../assets/icons/tiles.svg"
import { ReactComponent as RocketIcon } from "../../../assets/icons/rocket.svg"
import { ReactComponent as CloudIcon } from "../../../assets/icons/cloud.svg"
import { ReactComponent as CursorIcon } from "../../../assets/icons/cursor.svg"

import "./Features.sass"

const Features = () => {
  return (
    <div className="features">
      <div className="features__title">
        <span className="bold-text">Droplr</span> lets you
      </div>
      <div className="features__subtitle">
        Cut down on complexity and have the job done, simply
      </div>
      <div className="features__content">
        <Feature
          renderIcon={() => <CursorIcon className="feature__icon -green" />}
          title="Improve communication"
          subtitle="Pictures speak louder than words. Use that! Snap Screenshots
          and mark them up. Do you need to clarify a more complex
          problem? Record a Screen. Get your point across in a few clicks."
        />
        <Feature
          renderIcon={() => <RocketIcon className="feature__icon -red" />}
          title="Be more productive"
          subtitle="Don't let the lack of information slow you down. Use Droplr
          Integrations and make the flow of information seamless.
          Communicate as you go and finish your projects faster"
        />
        <Feature
          renderIcon={() => <CloudIcon className="feature__icon -blue" />}
          title="Stay organized"
          subtitle="Store your files in a cloud and have them always easily
          accessible. Create category Boards to avoid clutter. Droplr
          clean, easy to-use interface makes managing files hassle-free."
        />
        <Feature
          renderIcon={() => <TileIcon className="feature__icon -yellow" />}
          title="Collaborate effectively"
          subtitle="Empower whole teams to collaborate. Use Droplr Dashboard
          as a collaboration platform. Let others access and modify the
          content of your Boards so that everyone is up-to-speed."
        />
      </div>
      <a href="#features">Browse all awesome Droplr features</a>
    </div>
  )
}

export default Features
