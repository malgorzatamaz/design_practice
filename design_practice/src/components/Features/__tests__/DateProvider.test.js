import React from "react"
import { shallow, mount } from "enzyme"

import DateProvider from "../DateProvider.js"
import Features from "../Features.js"

describe("Features renders", () => {
  const wrapper = shallow(<Features />)

  it("should render date of my birth", () => {
    const wrapperRender = wrapper.find(DateProvider).renderProp("render")({
      day: 25,
      month: 11,
      year: 1994
    })

    expect(wrapperRender.text()).toMatchSnapshot()
  })

  it("should render current date", () => {
    const today = new Date()
    const date = {
      day: today.getDate(),
      month: today.getMonth() + 1,
      year: today.getFullYear()
    }

    const wrapperRender = wrapper.find(DateProvider).renderProp("render")(date)

    expect(
      wrapper
        .find(DateProvider)
        .dive()
        .text()
    ).toBe(`${date.day}-${date.month}-${date.year}`)
  })
})
