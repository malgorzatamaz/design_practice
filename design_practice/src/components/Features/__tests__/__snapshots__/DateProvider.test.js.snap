// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features renders should render date of my birth 1`] = `"25-11-1994"`;
